pkg/java/pom/settings.go

```diff
@@ -8,24 +8,43 @@ import (
 	"golang.org/x/net/html/charset"
 )
 
+type Server struct {
+	ID       string `xml:"id"`
+	Username string `xml:"username"`
+	Password string `xml:"password"`
+}
+
 type settings struct {
-	LocalRepository string `xml:"localRepository"`
+	LocalRepository string   `xml:"localRepository"`
+	Servers         []Server `xml:"servers>server"`
 }
 
 func readSettings() settings {
+	s := settings{}
+	settingsFound := false
+
 	userSettingsPath := filepath.Join(os.Getenv("HOME"), ".m2", "settings.xml")
 	userSettings, err := openSettings(userSettingsPath)
-	if err == nil && userSettings.LocalRepository != "" {
-		return userSettings
+	if err == nil {
+		if userSettings.LocalRepository != "" {
+			return userSettings
+		}
+		s = userSettings
+		settingsFound = true
 	}
 
 	globalSettingsPath := filepath.Join(os.Getenv("MAVEN_HOME"), "conf", "settings.xml")
 	globalSettings, err := openSettings(globalSettingsPath)
-	if err == nil && globalSettings.LocalRepository != "" {
-		return globalSettings
+	if err == nil {
+		if globalSettings.LocalRepository != "" {
+			return globalSettings
+		}
+		if !settingsFound {
+			s = globalSettings
+		}
 	}
```

Comment:
Looks like we need to merge global and user settings - `The former settings.xml are also called global settings, the latter settings.xml are referred to as user settings. If both files exists, their contents gets merged, with the user-specific settings.xml being dominant.` (https://maven.apache.org/settings.html#quick-overview)
order for `LocalRepository` : `user` -> `global`
But for repositories we should use `global` -> `user` order - https://maven.apache.org/guides/mini/guide-multiple-repositories.html#repository-order

Reply:
Thanks @DmitriyLewen for the review and pointers. I have updated the code based on your suggestion. Please take a look

Reply:
These changes seem good to me.
Some moments:
1 - Add comments with links to order, please.
2. I'm not sure if `/usr/share/maven` is default path. I didn't find any information on this path. Do you have docs on this matter?
If I remember correctly, you set MAVEN_HOME when you install maven. So MAVEN_HOME must exist.

Reply:
- Added comments and links to document
- Added testcase
- What I found is that MAVEN_HOME is not not mandatory and not always set. 
- `mvn` find the home based on how/where its installed unless MAVEN_HOME variable is set. 
- `/usr/share/maven` is not always the default path and it depends on how maven in installed. For Ubuntu installation with apt it is `/usr/share/maven` , but if we download the tar.gz from maven and install, it will be the extracted folder. 
```/usr/bin/mvn -V
Apache Maven 3.6.3
Maven home: /usr/share/maven

./mvn -V
Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)
Maven home: /home/reny/apache-maven-3.9.6```
So if we keep the default value as `/usr/share/maven` it will get the settings in some OS where if its not there, the settings will not get it anywhere. Let me know if you need the default value to be removed @DmitriyLewen